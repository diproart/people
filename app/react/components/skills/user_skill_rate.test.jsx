import React from 'react';
import UserSkillRate from './user_skill_rate';
import {shallow, mount } from 'enzyme';

describe('UserSkillRate', () => {

  beforeAll(() => {
    window.$ = function () {
      return {
        tooltip() {
          return 'test';
        }
      }
    }
  });

  const testingComponent = <UserSkillRate
    favorite={true}
    note={'test note'}
    rate={1}
    rate_type={'range'}
    name={'test skill name'}
    description={'test skill description'}
    id={12}
  />

  test('component structure match snapshot', () => {
    const subject = mount(testingComponent);
    expect(subject).toMatchSnapshot();
  });

  test('clicks on favorite element updates favorite state', () => {
    const subject = mount(testingComponent);

    expect(subject.state().favorite).toBe(true);
    subject.find('.skill__row__favorite-btn').first().simulate('click')
    expect(subject.state().favorite).toBe(false);
  });

  test('fills note in textarea to change note', () => {
    const subject = mount(testingComponent);
    expect(subject.state().note).toBe('test note');
    subject.find('.skill__note').node.value = 'new note'
    subject.find('.skill__note').simulate('change');
    expect(subject.state().note).toBe('new note');
  });

  test('triggers onRateChange updates rate state', () => {
    const subject = mount(testingComponent);
    expect(subject.state().rate).toBe(1);
    subject.node.onRateChange(2)
    expect(subject.state().rate).toBe(2);
  });
});
