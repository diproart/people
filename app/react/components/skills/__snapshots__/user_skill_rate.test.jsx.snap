exports[`UserSkillRate component structure match snapshot 1`] = `
<UserSkillRate
  description="test skill description"
  favorite={true}
  id={12}
  name="test skill name"
  note="test note"
  rate={1}
  rate_type="range">
  <tr
    className="skill__row">
    <td>
      test skill name
    </td>
    <td>
      test skill description
    </td>
    <td>
      <RateScale
        onRateChange={[Function]}
        rate={1}
        rateType="range">
        <ul
          className="list-inline skill__rating skill__rating--range">
          <li
            className="skill__clear_rate">
            <i
              className="glyphicon glyphicon-remove"
              data-placement="top"
              data-rate="0"
              data-toggle="tooltip"
              onClick={[Function]}
              title="Click to reset your rating" />
          </li>
          <li>
            <i
              className="glyphicon skill__rate skill__rate--range glyphicon-star selected"
              data-placement="top"
              data-rate={1}
              data-toggle="tooltip"
              onClick={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              title="m" />
          </li>
          <li>
            <i
              className="glyphicon skill__rate skill__rate--range glyphicon-star-empty"
              data-placement="top"
              data-rate={2}
              data-toggle="tooltip"
              onClick={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              title="i" />
          </li>
          <li>
            <i
              className="glyphicon skill__rate skill__rate--range glyphicon-star-empty"
              data-placement="top"
              data-rate={3}
              data-toggle="tooltip"
              onClick={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              title="s" />
          </li>
        </ul>
      </RateScale>
    </td>
    <td
      className="skill__row__favorite-btn"
      onClick={[Function]}>
      <i
        className="skill__favorite glyphicon glyphicon-heart selected"
        data-placement="top"
        data-toggle="tooltip"
        title="[missing \"en.skills.favorite\" translation]" />
    </td>
    <td>
      <textarea
        className="skill__note form-control"
        cols="30"
        onChange={[Function]}
        placeholder="[missing \"en.skills.add_note\" translation]"
        rows="1"
        value="test note" />
    </td>
  </tr>
</UserSkillRate>
`;
